Lib/defconQt/glyphCollectionView.py

```diff
@@ -229,31 +248,33 @@ def keyPressEvent(self, event):
         elif key == Qt.Key_Return:
             index = self._lastSelectedCell
             if index is not None and self.doubleClickCallback is not None:
-                # TODO: does it still make sense to call this doubleClickCallback?
+                # TODO: does it still make sense to call this
+                #       doubleClickCallback?
                 self.doubleClickCallback(self._glyphs[index])
         elif event.matches(QKeySequence.SelectAll):
             self.selection = set(range(len(self._glyphs)))
         elif key == Qt.Key_D and modifiers & Qt.ControlModifier:
             self.selection = set()
-        # XXX: this is specific to fontView so should be done thru subclassing of a base widget,
-        # as is done in groupsView
+        # XXX: this is specific to fontView so should be done thru subclassing
+        # of a base widget, as is done in groupsView
         elif key == platformSpecific.deleteKey:
-            #if self.characterDeletionCallback is not None:
+            # if self.characterDeletionCallback is not None:
             if proceedWithDeletion(self) and self.selection:
                 # we need to del in reverse order to keep key references valid
                 for key in sorted(self._selection, reverse=True):
                     glyph = self._glyphs[key]
-                    font = glyph.getParent()
+                    # font = glyph.getParent()  # unused
                     if modifiers & Qt.ShiftModifier:
-                        del self.font[gName]
+                        # del self.font[gName]
+                        # # gName is undefined
```

Comment:
Keep the font variable up here and replace self.font &rarr; font and gName &rarr; glyph.name.